src/main/kotlin/simplerag/ragback/domain/document/service/DataFileService.kt

```
@@ -54,25 +49,23 @@ class DataFileService(
                 throw FileException(ErrorCode.ALREADY_FILE, sha256)
             }
 
-            val fileUrl = s3Util.upload(file, S3Type.ORIGINAL_FILE)
-            uploadedUrls += fileUrl
+            val content = contentExtractorResolver.extractContent(file, type)
 
             val dataFile = try {
-                dataFileRepository.save(DataFile.from(meta.title, type, sizeBytes, sha256, fileUrl))
+                dataFileRepository.save(DataFile.from(meta.title, type, sizeBytes, sha256, content))
             } catch (ex: DataIntegrityViolationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)